walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,107 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
+
+  /*
+   Criterion functional interface is used to define who's (mandatory or all attendees')
+   occupied hours should be taken in account in findAvailableTimeRanges function
+   */
+  @FunctionalInterface
+  private interface Criterion {
+    boolean satisfies(int x);
+  }
+  private final Criterion COMMON_CRITERION = x -> x == 0;
+  private final Criterion MANDATORY_CRITERION = x -> x <= 1;
+
+  private static final int DURATION_DAY = 60 * 24;
```

Comment:
I suggest mentioning the unit of the duration, in this case minutes, in the variable name. With constants like that, if the code ages and someone sees it pop up in autocomplete, they should know which number to expect when they use it without having to jump to the definition.